pkg/strc/multi.go

```diff
@@ -134,8 +138,9 @@ func (h *MultiHandler) WithGroup(name string) slog.Handler {
 	}
 
 	return &MultiHandler{
-		handlers: handlers,
-		inGroup:  true,
+		handlers:    handlers,
+		callback:    h.callback,
+		inSpanGroup: name == SpanGroupName,
 	}
 }
 
```

Comment:
All these changes improves and simplifies the callback mechanism which is used to inject context values from HTTP headers.